packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -262,6 +270,24 @@ export const useGridKeyboardNavigation = (
     [apiRef, currentPage, goToCell, goToHeader],
   );
 
+  const handleCellKeyDown = React.useCallback<GridEventListener<GridEvents.cellKeyDown>>(
+    (params, event) => {
+      // Ignore portal
+      if (!event.currentTarget.contains(event.target as Element)) {
+        return;
+      }
+
+      // Get the most recent params because the cell mode may have changed by another listener
+      const cellParams = apiRef.current.getCellParams(params.id, params.field);
+
+      if (cellParams.cellMode !== GridCellModes.Edit && isNavigationKey(event.key)) {
```

Comment:
<kbd>Backspace</kbd> is not a navigation key if it is pressed while the focus is on the checkbox column. It jumps if a click only on the border of the checkbox. By doing an additional check here, you can remove the `event.preventDefault`. Also, the modification in the order that the hooks are called can be reverted.

![msedge_5cXogi4eMh](https://user-images.githubusercontent.com/42154031/159328837-36a1f640-ef03-4952-b450-ba3c966316f0.gif)